app/src/main/java/swati4star/createpdf/activity/FavouritesActivity.java

```diff
@@ -1,16 +1,122 @@
+import android.support.v7.preference.PreferenceManager;
+import android.support.v7.widget.Toolbar;
+import android.view.Menu;
+import android.view.MenuItem;
 
 import swati4star.createpdf.R;
 
+import static swati4star.createpdf.util.Constants.ADD_IMAGES_KEY;
+import static swati4star.createpdf.util.Constants.ADD_PASSWORD_KEY;
+import static swati4star.createpdf.util.Constants.ADD_WATERMARK_KEY;
+import static swati4star.createpdf.util.Constants.COMPRESS_PDF_KEY;
+import static swati4star.createpdf.util.Constants.EXTRACT_IMAGES_KEY;
+import static swati4star.createpdf.util.Constants.HISTORY_KEY;
+import static swati4star.createpdf.util.Constants.IMAGE_TO_PDF_KEY;
+import static swati4star.createpdf.util.Constants.INVERT_PDF_KEY;
+import static swati4star.createpdf.util.Constants.MERGE_PDF_KEY;
+import static swati4star.createpdf.util.Constants.PDF_TO_IMAGES_KEY;
+import static swati4star.createpdf.util.Constants.QR_BARCODE_KEY;
+import static swati4star.createpdf.util.Constants.REMOVE_DUPLICATE_PAGES_KEY;
+import static swati4star.createpdf.util.Constants.REMOVE_PAGES_KEY;
+import static swati4star.createpdf.util.Constants.REMOVE_PASSWORD_KEY;
+import static swati4star.createpdf.util.Constants.REORDER_PAGES_KEY;
+import static swati4star.createpdf.util.Constants.ROTATE_PAGES_KEY;
+import static swati4star.createpdf.util.Constants.SPLIT_PDF_KEY;
+import static swati4star.createpdf.util.Constants.TEXT_TO_PDF_KEY;
+import static swati4star.createpdf.util.Constants.VIEW_FILES_KEY;
+
 public class FavouritesActivity extends AppCompatActivity {
 
+    private SharedPreferences mSharedpreferences;
+    private boolean[] mKeyState = new boolean[19];
+
     @Override
     protected void onCreate(@Nullable Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        mSharedpreferences = PreferenceManager
+                .getDefaultSharedPreferences(this);
         setContentView(R.layout.fav_pref_screen);
+        storeInitialState();
+        Toolbar toolbar = findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
+        getSupportActionBar().setDisplayShowHomeEnabled(true);
+    }
+
+    @Override
+    public boolean onCreateOptionsMenu(Menu menu) {
+        getMenuInflater().inflate(R.menu.menu_favourite_pref_done, menu);
+        return true;
+    }
+
+    @Override
+    public boolean onOptionsItemSelected(MenuItem item) {
+        switch (item.getItemId()) {
+            case android.R.id.home:
+                onBackPressed();
+                return true;
+            case R.id.fav_action_done:
+                super.onBackPressed();
+                return true;
+            default:
+                return super.onOptionsItemSelected(item);
+        }
+    }
+
+    @Override
+    public void onBackPressed() {
+        onBackPressedState();
+        super.onBackPressed();
+    }
+
+
+    private void storeInitialState() {
```

Comment:
addd javadoc style comments to the function

Reply:
Okay